homeassistant/components/shelly/entity.py

```diff
@@ -187,6 +234,23 @@ class BlockAttributeDescription:
     extra_state_attributes: Callable[[Block], dict | None] | None = None
 
 
+@dataclass
+class RpcAttributeDescription:
```

Comment:
Can we consider inheriting from `SensorEntityDescription ` ?

Reply:
I have started with `SensorEntityDescription` since we later make two type of entities from it for `sensor` and `binary_sensor` the code becomes a mess, also we have too much customizations we already did in our on `AttributeDescription` class the code based on `SensorEntityDescription` had about 3X lines more than current solution, we can think of a better way as a general code quality improvement, but it should take into consideration all the other types of `EntityDescription` classes we have.